The-Puffin-Room/src/components/ArticleByTopics.jsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState, useCallback } from "react";
+import { useParams } from "react-router-dom";
+import ArticleCard from "./ArticleCard.jsx";
+import Loading from "./Loading.jsx";
+import {getArticleByTopic } from "../api.js";
+
+function ArticleByTopics() {
+  const [articles, setArticles] = useState([]);
+  const [page, setPage] = useState(1);
+  const [isLoading, setIsLoading] = useState(false);
+  const [totalPages, setTotalPages] = useState(1);
+
+  const { topic } = useParams();
+  const reformattedTopic = topic.charAt(0).toUpperCase() + topic.slice(1);
+
+  const getArticles = useCallback(() => {
+    setIsLoading(true);
+    getArticleByTopic(topic)
+      .then((allArticles) => {
+        setArticles(allArticles);
+        setTotalPages(Math.ceil(allArticles[0].total_count / 10));
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.error("Error fetching articles:", error);
+        setIsLoading(false);
+      });
+  }, [page,topic]);
+
+  useEffect(() => {
+    getArticles();
+  }, [getArticles]);
+
+  const nextPage = () => {
+    if (page < totalPages) setPage(page + 1);
+  };
+
+  const prevPage = () => {
+    if (page > 1) setPage(page - 1);
+  };
+
+  return (
+    <>
+      <h2>{reformattedTopic}</h2>
+      {isLoading ? (
+        <Loading isLoading={isLoading} />
```

Comment:
Surely if we make it down to line 46 isLoading will always be true - so we don't need to pass it into the component?